app/src/main/java/com/github/se/eventradar/ui/home/Home.kt

```diff
@@ -60,6 +58,11 @@ fun HomeScreen(
   // Ui States handled by viewModel
   val uiState by viewModel.uiState.collectAsState()
   LaunchedEffect(key1 = uiState.eventList) { viewModel.getEvents() }
+  LaunchedEffect(key1 = uiState.tab) {
+    if (uiState.tab == Tab.UPCOMING) {
+      viewModel.getUpcomingEvents()
+    }
+  }
```

Comment:
concern: If getUpcomingEvents() updates the uiState.eventList, and getEvents() is triggered whenever uiState.eventList changes, there's a possibility of an unintended recursion or multiple calls to getEvents().

Reply:
Yes. Will update this.